src/app/layout.tsx

```diff
@@ -0,0 +1,40 @@
+import "./app.css";
+import StoreProvider from "@/store/storeProviders";
+import { AppProvider } from "@/app/page/[pageId]/AppProvider";
+
+import type { Metadata } from "next";
+import { ThemeProvider } from "@/utils/ThemeProvider";
+import { Geist, Geist_Mono } from "next/font/google";
+import "./index.css";
+import "./app.css";
+
+const geistSans = Geist({
+  variable: "--font-geist-sans",
+  subsets: ["latin"],
+});
+
+const geistMono = Geist_Mono({
+  variable: "--font-geist-mono",
+  subsets: ["latin"],
+});
+
+export const metadata: Metadata = {
+  title: "RS Next App",
+  description: "Rick and Morty Next.js App",
+};
+
+export default function RootLayout({ children }: { children: React.ReactNode }) {
+  return (
+    <html lang="en">
+      <body className={`${geistSans.variable} ${geistMono.variable}`}>
+        <StoreProvider>
+          <ThemeProvider>
+            <AppProvider>
+              <main id="root">{children}</main>
+            </AppProvider>
+          </ThemeProvider>
+        </StoreProvider>
```

Comment:
It might be helpful to extract all providers into a single component to simplify `RootLayout` and better manage global providers.
```suggestion
        <AppProviders>
          <main id="root">{children}</main>
        </AppProviders>
        
 // src/components/AppProviders.tsx 
import { ReactNode } from "react";
import StoreProvider from "@/store/storeProviders";
import { AppProvider } from "@/app/page/[pageId]/AppProvider";
import { ThemeProvider } from "@/utils/ThemeProvider";

export default function AppProviders({ children }: { children: ReactNode }) {
  return (
    <StoreProvider>
      <ThemeProvider>
        <AppProvider>{children}</AppProvider>
      </ThemeProvider>
    </StoreProvider>
  );
}
```